presto-main/src/main/java/io/prestosql/operator/NestedLoopJoinOperator.java

```diff
@@ -123,21 +121,19 @@ public OperatorFactory duplicate()
     private final OperatorContext operatorContext;
     private final Runnable afterClose;
 
-    private final List<Integer> probeChannels;
```

Comment:
These channels should be used for reordering only and not pruning. If we use this for (build) prunning, we lost already because data was already transferred over the network.

I think we should have something plan validator like `ValidateCrossJoins` that would assert that cross joins do not do build pruning.

Reply:
Cross join should never have to prune inputs from neither source. If it did, it would fail in `ValidateDependenciesChecker`:
```
checkDependencies(node.getOutputSymbols(), inputs, "Cross join output symbols (%s) must contain all of the source symbols (%s)", node.getOutputSymbols(), inputs);
```